src/pretix/base/models/vouchers.py

```diff
@@ -13,8 +13,10 @@
 from .items import Item, ItemVariation, Quota
 
 
-def _generate_random_code():
+def _generate_random_code(prefix=None):
     charset = list('ABCDEFGHKLMNPQRSTUVWXYZ23456789')
+    if prefix:
+        return prefix + get_random_string(length=settings.ENTROPY['voucher_code'] - len(prefix), allowed_chars=charset)
```

Comment:
I'm not sure if we should subtract here or just keep the length. The default length is only 16 characters, so entropy gets quite low with a medium-long prefix already.